templates/clusterclass-template.yaml

```diff
@@ -0,0 +1,625 @@
+        valueFrom:
+          template: |
+            - name: root
+              sshAuthorizedKeys:
+              - '{{ .sshKey }}'
+      selector:
+        apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+        kind: KubeadmConfigTemplate
+        matchResources:
+          machineDeploymentClass:
+            names:
+            - ${CLUSTER_CLASS_NAME}-worker
+    enabledIf: '{{ if .sshKey }}true{{end}}'
+    name: enableSSHIntoNodes
+  - definitions:
+    - jsonPatches:
+      - op: add
+        path: /spec/template/spec/kubeadmConfigSpec/files/-
+        valueFrom:
+          template: |
+            # CSI Metadata config
+            content: |
+              {
+                "datacenter-id": "{{ .datacenterID }}"
+              }
+            owner: root:root
+            path: /etc/ie-csi/cfg.json
+            permissions: '0644'
+      selector:
+        apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+        kind: KubeadmControlPlaneTemplate
+        matchResources:
+          controlPlane: true
+    - jsonPatches:
+      - op: add
+        path: /spec/template/spec/files/-
+        valueFrom:
+          template: |
+            # CSI Metadata config
+            content: |
+              {
+                "datacenter-id": "{{ .datacenterID }}"
+              }
+            owner: root:root
+            path: /etc/ie-csi/cfg.json
+            permissions: '0644'
+      selector:
+        apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+        kind: KubeadmConfigTemplate
+        matchResources:
+          machineDeploymentClass:
+            names:
+            - ${CLUSTER_CLASS_NAME}-worker
+    - jsonPatches:
+      - op: add
+        path: /spec/template/spec/datacenterID
+        valueFrom:
+          variable: datacenterID
+      selector:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha1
+        kind: IonosCloudMachineTemplate
+        matchResources:
+          controlPlane: true
+    - jsonPatches:
+      - op: add
+        path: /spec/template/spec/datacenterID
+        valueFrom:
+          variable: datacenterID
+      selector:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha1
+        kind: IonosCloudMachineTemplate
+        matchResources:
+          machineDeploymentClass:
+            names:
+            - ${CLUSTER_CLASS_NAME}-worker
+    name: datacenterIdSubstitutionsOnKubeadm
+  - definitions:
+    - jsonPatches:
+      - op: add
+        path: /spec/template/spec/kubeadmConfigSpec/initConfiguration/localAPIEndpoint/bindPort
+        valueFrom:
+          variable: controlPlanePort
+      selector:
+        apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+        kind: KubeadmControlPlaneTemplate
+        matchResources:
+          controlPlane: true
+    - jsonPatches:
+      - op: add
+        path: /spec/template/spec/controlPlaneEndpoint
+        valueFrom:
+          template: |
+            host: '{{ .controlPlaneHost | default .controlPlaneIpAddr }}'
```

Comment:
Since both are of type string, you could simply collapse these two into one value (a host can be an ip address).

Reply:
No. The ip is required in kubevip as the host can be an fqdn.